lib/mpi/messages/find_profile_by_facility.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_relative 'request_helper'
+require_relative 'request_builder'
+require 'mpi/constants'
+
+module MPI
+  module Messages
+    class FindProfileByFacility
+      attr_reader :facility_num, :dfn, :search_type
+
+      def initialize(facility_num:, dfn:, search_type: MPI::Constants::CORRELATION_WITH_RELATIONSHIP_DATA)
+        @facility_num = facility_num
+        @dfn = dfn
+        @search_type = search_type
+      end
+
+      def perform
+        MPI::Messages::RequestBuilder.new(extension: MPI::Constants::FIND_PROFILE, body: build_body).perform
+      rescue => e
+        Rails.logger.error "[FindProfileByFacility] Failed to build request: #{e.message}"
+        raise e
+      end
+
+      private
+
+      def build_body
+        body = RequestHelper.build_control_act_process_element
+        body << RequestHelper.build_code(code: MPI::Constants::FIND_PROFILE_CONTROL_ACT_PROCESS)
+        body << query_by_parameter
+        body
+      end
+
+      def query_by_parameter
+        query_by_parameter = RequestHelper.build_query_by_parameter(search_type:)
+        query_by_parameter << build_parameter_list
+      end
+
+      def build_parameter_list
+        element = RequestHelper.build_parameter_list_element
+        identifier = "#{dfn}^PI^#{facility_num}^#{MPI::Constants::VHA_FULL_IDENTIFIER}"
```

Comment:
I might suggest making `identifier` a function like `root`, just a bit of consistency but a nitpick regardless